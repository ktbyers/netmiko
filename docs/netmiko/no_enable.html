<!doctype html>
<html lang="en">
<head>
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, minimum-scale=1" />
<meta name="generator" content="pdoc 0.10.0" />
<title>netmiko.no_enable API documentation</title>
<meta name="description" content="" />
<link rel="preload stylesheet" as="style" href="https://cdnjs.cloudflare.com/ajax/libs/10up-sanitize.css/11.0.1/sanitize.min.css" integrity="sha256-PK9q560IAAa6WVRRh76LtCaI8pjTJ2z11v0miyNNjrs=" crossorigin>
<link rel="preload stylesheet" as="style" href="https://cdnjs.cloudflare.com/ajax/libs/10up-sanitize.css/11.0.1/typography.min.css" integrity="sha256-7l/o7C8jubJiy74VsKTidCy1yBkRtiUGbVkYBylBqUg=" crossorigin>
<link rel="stylesheet preload" as="style" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/10.1.1/styles/github.min.css" crossorigin>
<style>:root{--highlight-color:#fe9}.flex{display:flex !important}body{line-height:1.5em}#content{padding:20px}#sidebar{padding:30px;overflow:hidden}#sidebar > *:last-child{margin-bottom:2cm}.http-server-breadcrumbs{font-size:130%;margin:0 0 15px 0}#footer{font-size:.75em;padding:5px 30px;border-top:1px solid #ddd;text-align:right}#footer p{margin:0 0 0 1em;display:inline-block}#footer p:last-child{margin-right:30px}h1,h2,h3,h4,h5{font-weight:300}h1{font-size:2.5em;line-height:1.1em}h2{font-size:1.75em;margin:1em 0 .50em 0}h3{font-size:1.4em;margin:25px 0 10px 0}h4{margin:0;font-size:105%}h1:target,h2:target,h3:target,h4:target,h5:target,h6:target{background:var(--highlight-color);padding:.2em 0}a{color:#058;text-decoration:none;transition:color .3s ease-in-out}a:hover{color:#e82}.title code{font-weight:bold}h2[id^="header-"]{margin-top:2em}.ident{color:#900}pre code{background:#f8f8f8;font-size:.8em;line-height:1.4em}code{background:#f2f2f1;padding:1px 4px;overflow-wrap:break-word}h1 code{background:transparent}pre{background:#f8f8f8;border:0;border-top:1px solid #ccc;border-bottom:1px solid #ccc;margin:1em 0;padding:1ex}#http-server-module-list{display:flex;flex-flow:column}#http-server-module-list div{display:flex}#http-server-module-list dt{min-width:10%}#http-server-module-list p{margin-top:0}.toc ul,#index{list-style-type:none;margin:0;padding:0}#index code{background:transparent}#index h3{border-bottom:1px solid #ddd}#index ul{padding:0}#index h4{margin-top:.6em;font-weight:bold}@media (min-width:200ex){#index .two-column{column-count:2}}@media (min-width:300ex){#index .two-column{column-count:3}}dl{margin-bottom:2em}dl dl:last-child{margin-bottom:4em}dd{margin:0 0 1em 3em}#header-classes + dl > dd{margin-bottom:3em}dd dd{margin-left:2em}dd p{margin:10px 0}.name{background:#eee;font-weight:bold;font-size:.85em;padding:5px 10px;display:inline-block;min-width:40%}.name:hover{background:#e0e0e0}dt:target .name{background:var(--highlight-color)}.name > span:first-child{white-space:nowrap}.name.class > span:nth-child(2){margin-left:.4em}.inherited{color:#999;border-left:5px solid #eee;padding-left:1em}.inheritance em{font-style:normal;font-weight:bold}.desc h2{font-weight:400;font-size:1.25em}.desc h3{font-size:1em}.desc dt code{background:inherit}.source summary,.git-link-div{color:#666;text-align:right;font-weight:400;font-size:.8em;text-transform:uppercase}.source summary > *{white-space:nowrap;cursor:pointer}.git-link{color:inherit;margin-left:1em}.source pre{max-height:500px;overflow:auto;margin:0}.source pre code{font-size:12px;overflow:visible}.hlist{list-style:none}.hlist li{display:inline}.hlist li:after{content:',\2002'}.hlist li:last-child:after{content:none}.hlist .hlist{display:inline;padding-left:1em}img{max-width:100%}td{padding:0 .5em}.admonition{padding:.1em .5em;margin-bottom:1em}.admonition-title{font-weight:bold}.admonition.note,.admonition.info,.admonition.important{background:#aef}.admonition.todo,.admonition.versionadded,.admonition.tip,.admonition.hint{background:#dfd}.admonition.warning,.admonition.versionchanged,.admonition.deprecated{background:#fd4}.admonition.error,.admonition.danger,.admonition.caution{background:lightpink}</style>
<style media="screen and (min-width: 700px)">@media screen and (min-width:700px){#sidebar{width:30%;height:100vh;overflow:auto;position:sticky;top:0}#content{width:70%;max-width:100ch;padding:3em 4em;border-left:1px solid #ddd}pre code{font-size:1em}.item .name{font-size:1em}main{display:flex;flex-direction:row-reverse;justify-content:flex-end}.toc ul ul,#index ul{padding-left:1.5em}.toc > ul > li{margin-top:.5em}}</style>
<style media="print">@media print{#sidebar h1{page-break-before:always}.source{display:none}}@media print{*{background:transparent !important;color:#000 !important;box-shadow:none !important;text-shadow:none !important}a[href]:after{content:" (" attr(href) ")";font-size:90%}a[href][title]:after{content:none}abbr[title]:after{content:" (" attr(title) ")"}.ir a:after,a[href^="javascript:"]:after,a[href^="#"]:after{content:""}pre,blockquote{border:1px solid #999;page-break-inside:avoid}thead{display:table-header-group}tr,img{page-break-inside:avoid}img{max-width:100% !important}@page{margin:0.5cm}p,h2,h3{orphans:3;widows:3}h1,h2,h3,h4,h5,h6{page-break-after:avoid}}</style>
<script defer src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/10.1.1/highlight.min.js" integrity="sha256-Uv3H6lx7dJmRfRvH8TH6kJD1TSK1aFcwgx+mdg3epi8=" crossorigin></script>
<script>window.addEventListener('DOMContentLoaded', () => hljs.initHighlighting())</script>
</head>
<body>
<main>
<article id="content">
<header>
<h1 class="title">Module <code>netmiko.no_enable</code></h1>
</header>
<section id="section-intro">
<details class="source">
<summary>
<span>Expand source code</span>
</summary>
<pre><code class="python">from typing import Optional
import re


class NoEnable:
    &#34;&#34;&#34;
    Class for platforms that have no enable mode.

    Netmiko translates the meaning of &#34;enable&#34; mode to be a proxy for &#34;can
    go into config mode&#34;. In other words, that you ultimately have privileges
    to execute configuration changes.

    The expectation on platforms that have no method for elevating privileges
    is that the standard default privileges allow configuration changes.

    Consequently check_enable_mode returns True by default for platforms that
    don&#39;t explicitly support enable mode.
    &#34;&#34;&#34;

    def check_enable_mode(self, check_string: str = &#34;&#34;) -&gt; bool:
        return True

    def enable(
        self,
        cmd: str = &#34;&#34;,
        pattern: str = &#34;&#34;,
        enable_pattern: Optional[str] = None,
        re_flags: int = re.IGNORECASE,
    ) -&gt; str:
        return &#34;&#34;

    def exit_enable_mode(self, exit_command: str = &#34;&#34;) -&gt; str:
        return &#34;&#34;</code></pre>
</details>
</section>
<section>
</section>
<section>
</section>
<section>
</section>
<section>
<h2 class="section-title" id="header-classes">Classes</h2>
<dl>
<dt id="netmiko.no_enable.NoEnable"><code class="flex name class">
<span>class <span class="ident">NoEnable</span></span>
</code></dt>
<dd>
<div class="desc"><p>Class for platforms that have no enable mode.</p>
<p>Netmiko translates the meaning of "enable" mode to be a proxy for "can
go into config mode". In other words, that you ultimately have privileges
to execute configuration changes.</p>
<p>The expectation on platforms that have no method for elevating privileges
is that the standard default privileges allow configuration changes.</p>
<p>Consequently check_enable_mode returns True by default for platforms that
don't explicitly support enable mode.</p></div>
<details class="source">
<summary>
<span>Expand source code</span>
</summary>
<pre><code class="python">class NoEnable:
    &#34;&#34;&#34;
    Class for platforms that have no enable mode.

    Netmiko translates the meaning of &#34;enable&#34; mode to be a proxy for &#34;can
    go into config mode&#34;. In other words, that you ultimately have privileges
    to execute configuration changes.

    The expectation on platforms that have no method for elevating privileges
    is that the standard default privileges allow configuration changes.

    Consequently check_enable_mode returns True by default for platforms that
    don&#39;t explicitly support enable mode.
    &#34;&#34;&#34;

    def check_enable_mode(self, check_string: str = &#34;&#34;) -&gt; bool:
        return True

    def enable(
        self,
        cmd: str = &#34;&#34;,
        pattern: str = &#34;&#34;,
        enable_pattern: Optional[str] = None,
        re_flags: int = re.IGNORECASE,
    ) -&gt; str:
        return &#34;&#34;

    def exit_enable_mode(self, exit_command: str = &#34;&#34;) -&gt; str:
        return &#34;&#34;</code></pre>
</details>
<h3>Subclasses</h3>
<ul class="hlist">
<li><a title="netmiko.accedian.accedian_ssh.AccedianSSH" href="accedian/accedian_ssh.html#netmiko.accedian.accedian_ssh.AccedianSSH">AccedianSSH</a></li>
<li><a title="netmiko.alcatel.alcatel_aos_ssh.AlcatelAosSSH" href="alcatel/alcatel_aos_ssh.html#netmiko.alcatel.alcatel_aos_ssh.AlcatelAosSSH">AlcatelAosSSH</a></li>
<li><a title="netmiko.audiocode.audiocode_ssh.AudiocodeShellBase" href="audiocode/audiocode_ssh.html#netmiko.audiocode.audiocode_ssh.AudiocodeShellBase">AudiocodeShellBase</a></li>
<li><a title="netmiko.brocade.brocade_fos_ssh.BrocadeFOSSSH" href="brocade/brocade_fos_ssh.html#netmiko.brocade.brocade_fos_ssh.BrocadeFOSSSH">BrocadeFOSSSH</a></li>
<li><a title="netmiko.cdot.cdot_cros_ssh.CdotCrosSSH" href="cdot/cdot_cros_ssh.html#netmiko.cdot.cdot_cros_ssh.CdotCrosSSH">CdotCrosSSH</a></li>
<li><a title="netmiko.ciena.ciena_saos.CienaSaosBase" href="ciena/ciena_saos.html#netmiko.ciena.ciena_saos.CienaSaosBase">CienaSaosBase</a></li>
<li><a title="netmiko.cisco.cisco_ftd_ssh.CiscoFtdSSH" href="cisco/cisco_ftd_ssh.html#netmiko.cisco.cisco_ftd_ssh.CiscoFtdSSH">CiscoFtdSSH</a></li>
<li><a title="netmiko.coriant.coriant_ssh.CoriantSSH" href="coriant/coriant_ssh.html#netmiko.coriant.coriant_ssh.CoriantSSH">CoriantSSH</a></li>
<li><a title="netmiko.dell.dell_sonic_ssh.DellSonicSSH" href="dell/dell_sonic_ssh.html#netmiko.dell.dell_sonic_ssh.DellSonicSSH">DellSonicSSH</a></li>
<li><a title="netmiko.dlink.dlink_ds.DlinkDSBase" href="dlink/dlink_ds.html#netmiko.dlink.dlink_ds.DlinkDSBase">DlinkDSBase</a></li>
<li><a title="netmiko.extreme.extreme_nos_ssh.ExtremeNosSSH" href="extreme/extreme_nos_ssh.html#netmiko.extreme.extreme_nos_ssh.ExtremeNosSSH">ExtremeNosSSH</a></li>
<li><a title="netmiko.extreme.extreme_slx_ssh.ExtremeSlxSSH" href="extreme/extreme_slx_ssh.html#netmiko.extreme.extreme_slx_ssh.ExtremeSlxSSH">ExtremeSlxSSH</a></li>
<li><a title="netmiko.extreme.extreme_tierraos_ssh.ExtremeTierraSSH" href="extreme/extreme_tierraos_ssh.html#netmiko.extreme.extreme_tierraos_ssh.ExtremeTierraSSH">ExtremeTierraSSH</a></li>
<li><a title="netmiko.flexvnf.flexvnf_ssh.FlexvnfSSH" href="flexvnf/flexvnf_ssh.html#netmiko.flexvnf.flexvnf_ssh.FlexvnfSSH">FlexvnfSSH</a></li>
<li><a title="netmiko.huawei.huawei.HuaweiBase" href="huawei/huawei.html#netmiko.huawei.huawei.HuaweiBase">HuaweiBase</a></li>
<li><a title="netmiko.juniper.juniper.JuniperBase" href="juniper/juniper.html#netmiko.juniper.juniper.JuniperBase">JuniperBase</a></li>
<li><a title="netmiko.juniper.juniper_screenos.JuniperScreenOsSSH" href="juniper/juniper_screenos.html#netmiko.juniper.juniper_screenos.JuniperScreenOsSSH">JuniperScreenOsSSH</a></li>
<li><a title="netmiko.keymile.keymile_ssh.KeymileSSH" href="keymile/keymile_ssh.html#netmiko.keymile.keymile_ssh.KeymileSSH">KeymileSSH</a></li>
<li><a title="netmiko.mikrotik.mikrotik_ssh.MikrotikBase" href="mikrotik/mikrotik_ssh.html#netmiko.mikrotik.mikrotik_ssh.MikrotikBase">MikrotikBase</a></li>
<li><a title="netmiko.netapp.netapp_cdot_ssh.NetAppcDotSSH" href="netapp/netapp_cdot_ssh.html#netmiko.netapp.netapp_cdot_ssh.NetAppcDotSSH">NetAppcDotSSH</a></li>
<li><a title="netmiko.nokia.nokia_srl.NokiaSrlSSH" href="nokia/nokia_srl.html#netmiko.nokia.nokia_srl.NokiaSrlSSH">NokiaSrlSSH</a></li>
<li><a title="netmiko.paloalto.paloalto_panos.PaloAltoPanosBase" href="paloalto/paloalto_panos.html#netmiko.paloalto.paloalto_panos.PaloAltoPanosBase">PaloAltoPanosBase</a></li>
<li><a title="netmiko.rad.rad_etx.RadETXBase" href="rad/rad_etx.html#netmiko.rad.rad_etx.RadETXBase">RadETXBase</a></li>
<li><a title="netmiko.sixwind.sixwind_os.SixwindOSBase" href="sixwind/sixwind_os.html#netmiko.sixwind.sixwind_os.SixwindOSBase">SixwindOSBase</a></li>
<li><a title="netmiko.sophos.sophos_sfos_ssh.SophosSfosSSH" href="sophos/sophos_sfos_ssh.html#netmiko.sophos.sophos_sfos_ssh.SophosSfosSSH">SophosSfosSSH</a></li>
<li><a title="netmiko.supermicro.smci_smis.SmciSwitchSmisBase" href="supermicro/smci_smis.html#netmiko.supermicro.smci_smis.SmciSwitchSmisBase">SmciSwitchSmisBase</a></li>
<li><a title="netmiko.vyos.vyos_ssh.VyOSSSH" href="vyos/vyos_ssh.html#netmiko.vyos.vyos_ssh.VyOSSSH">VyOSSSH</a></li>
<li><a title="netmiko.zyxel.zyxel_ssh.ZyxelSSH" href="zyxel/zyxel_ssh.html#netmiko.zyxel.zyxel_ssh.ZyxelSSH">ZyxelSSH</a></li>
</ul>
<h3>Methods</h3>
<dl>
<dt id="netmiko.no_enable.NoEnable.check_enable_mode"><code class="name flex">
<span>def <span class="ident">check_enable_mode</span></span>(<span>self, check_string: str = '') ‑> bool</span>
</code></dt>
<dd>
<div class="desc"></div>
<details class="source">
<summary>
<span>Expand source code</span>
</summary>
<pre><code class="python">def check_enable_mode(self, check_string: str = &#34;&#34;) -&gt; bool:
    return True</code></pre>
</details>
</dd>
<dt id="netmiko.no_enable.NoEnable.enable"><code class="name flex">
<span>def <span class="ident">enable</span></span>(<span>self, cmd: str = '', pattern: str = '', enable_pattern: Optional[str] = None, re_flags: int = re.IGNORECASE) ‑> str</span>
</code></dt>
<dd>
<div class="desc"></div>
<details class="source">
<summary>
<span>Expand source code</span>
</summary>
<pre><code class="python">def enable(
    self,
    cmd: str = &#34;&#34;,
    pattern: str = &#34;&#34;,
    enable_pattern: Optional[str] = None,
    re_flags: int = re.IGNORECASE,
) -&gt; str:
    return &#34;&#34;</code></pre>
</details>
</dd>
<dt id="netmiko.no_enable.NoEnable.exit_enable_mode"><code class="name flex">
<span>def <span class="ident">exit_enable_mode</span></span>(<span>self, exit_command: str = '') ‑> str</span>
</code></dt>
<dd>
<div class="desc"></div>
<details class="source">
<summary>
<span>Expand source code</span>
</summary>
<pre><code class="python">def exit_enable_mode(self, exit_command: str = &#34;&#34;) -&gt; str:
    return &#34;&#34;</code></pre>
</details>
</dd>
</dl>
</dd>
</dl>
</section>
</article>
<nav id="sidebar">
<h1>Index</h1>
<div class="toc">
<ul></ul>
</div>
<ul id="index">
<li><h3>Super-module</h3>
<ul>
<li><code><a title="netmiko" href="index.html">netmiko</a></code></li>
</ul>
</li>
<li><h3><a href="#header-classes">Classes</a></h3>
<ul>
<li>
<h4><code><a title="netmiko.no_enable.NoEnable" href="#netmiko.no_enable.NoEnable">NoEnable</a></code></h4>
<ul class="">
<li><code><a title="netmiko.no_enable.NoEnable.check_enable_mode" href="#netmiko.no_enable.NoEnable.check_enable_mode">check_enable_mode</a></code></li>
<li><code><a title="netmiko.no_enable.NoEnable.enable" href="#netmiko.no_enable.NoEnable.enable">enable</a></code></li>
<li><code><a title="netmiko.no_enable.NoEnable.exit_enable_mode" href="#netmiko.no_enable.NoEnable.exit_enable_mode">exit_enable_mode</a></code></li>
</ul>
</li>
</ul>
</li>
</ul>
</nav>
</main>
<footer id="footer">
<p>Generated by <a href="https://pdoc3.github.io/pdoc" title="pdoc: Python API documentation generator"><cite>pdoc</cite> 0.10.0</a>.</p>
</footer>
</body>
</html>