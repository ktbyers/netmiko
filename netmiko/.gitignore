base_connection.py-backup cisco_base_connection.py-backup __init__.pyc-backup ssh_dispatcher.py-backup ssh_dispatcher.pyc-backup
