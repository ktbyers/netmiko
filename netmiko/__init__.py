from __future__ import unicode_literals
import logging

# Logging configuration
log = logging.getLogger(__name__) # noqa
log.addHandler(logging.NullHandler()) # noqa

from netmiko.ssh_dispatcher import ConnectHandler
from netmiko.ssh_dispatcher import ssh_dispatcher
from netmiko.ssh_dispatcher import redispatch
from netmiko.ssh_dispatcher import platforms
from netmiko.ssh_dispatcher import FileTransfer
from netmiko.scp_handler import SCPConn
from netmiko.cisco.cisco_ios import InLineTransfer
from netmiko.ssh_exception import NetMikoTimeoutException
from netmiko.ssh_exception import NetMikoAuthenticationException
from netmiko.ssh_autodetect import SSHDetect
from netmiko.base_connection import BaseConnection
from netmiko.scp_functions import file_transfer

# Alternate naming
NetmikoTimeoutError = NetMikoTimeoutException
NetmikoAuthError = NetMikoAuthenticationException
Netmiko = ConnectHandler

__version__ = '2.1.1'
__all__ = ('ConnectHandler', 'ssh_dispatcher', 'platforms', 'SCPConn', 'FileTransfer',
           'NetMikoTimeoutException', 'NetMikoAuthenticationException',
           'NetmikoTimeoutError', 'NetmikoAuthError', 'InLineTransfer', 'redispatch',
           'SSHDetect', 'BaseConnection', 'Netmiko', 'file_transfer')

# Cisco cntl-shift-six sequence
CNTL_SHIFT_6 = chr(30)
