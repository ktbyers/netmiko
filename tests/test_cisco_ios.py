#!/usr/bin/env python

import pytest

import netmiko
import time
from DEVICE_CREDS import *


def setup_module(module):

    module.EXPECTED_RESPONSES = {
        'base_prompt' : 'pynet-rtr1',
        'interface_ip'  : '10.220.88.20'
    }
    
    show_ver_command = 'show version'
    module.basic_command = 'show ip int brief'
    
    SSHClass = netmiko.ssh_dispatcher(cisco_881['device_type'])
    net_connect = SSHClass(**cisco_881)
    module.show_version = net_connect.send_command(show_ver_command)
    module.show_ip = net_connect.send_command(module.basic_command)
    module.base_prompt = net_connect.base_prompt

    # Test buffer clearing
    net_connect.remote_conn.send(show_ver_command)
    time.sleep(2)
    net_connect.clear_buffer()
    # Should not be anything there on the second pass
    module.clear_buffer_check = net_connect.clear_buffer()
    

def test_disable_paging():
    '''
    Verify paging is disabled by looking for string after when paging would
    normally occur
    '''
    assert 'Configuration register is' in show_version


def test_verify_ssh_connect():
    '''
    Verify the connection was established successfully
    '''
    assert 'Cisco IOS Software' in show_version


def test_verify_send_command():
    '''
    Verify a command can be sent down the channel successfully
    '''
    assert EXPECTED_RESPONSES['interface_ip'] in show_ip


def test_base_prompt():
    '''
    Verify the router prompt is detected correctly
    '''
    assert base_prompt == EXPECTED_RESPONSES['base_prompt']


def test_strip_prompt():
    '''
    Ensure the router prompt is not in the command output
    '''
    assert EXPECTED_RESPONSES['base_prompt'] not in show_ip


def test_strip_command():
    '''
    Ensure that the command that was executed does not show up in the 
    command output
    '''
    assert basic_command not in show_ip


def test_normalize_linefeeds():
    '''
    Ensure no '\r\n' sequences
    '''
    assert not '\r\n' in show_version


def test_clear_buffer():
    '''
    Test that clearing the buffer works
    '''
    assert clear_buffer_check is None
