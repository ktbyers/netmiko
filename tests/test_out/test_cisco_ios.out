Starting tests...good luck:
Cisco IOS
============================= test session starts ==============================
platform linux -- Python 3.8.5, pytest-5.1.2, py-1.10.0, pluggy-0.13.1 -- /home/kbyers/netmiko/.venv/bin/python3.8
cachedir: .pytest_cache
rootdir: /home/kbyers/netmiko, inifile: setup.cfg
plugins: pylama-7.7.1
collecting ... collected 20 items

test_netmiko_show.py::test_disable_paging PASSED
test_netmiko_show.py::test_terminal_width PASSED
test_netmiko_show.py::test_ssh_connect PASSED
test_netmiko_show.py::test_ssh_connect_cm PASSED
test_netmiko_show.py::test_send_command_timing PASSED
test_netmiko_show.py::test_send_command_timing_no_cmd_verify SKIPPED
test_netmiko_show.py::test_send_command PASSED
test_netmiko_show.py::test_send_command_no_cmd_verify SKIPPED
test_netmiko_show.py::test_complete_on_space_disabled SKIPPED
test_netmiko_show.py::test_send_command_textfsm PASSED
test_netmiko_show.py::test_send_command_ttp PASSED
test_netmiko_show.py::test_send_command_genie PASSED
test_netmiko_show.py::test_base_prompt PASSED
test_netmiko_show.py::test_strip_prompt PASSED
test_netmiko_show.py::test_strip_command PASSED
test_netmiko_show.py::test_normalize_linefeeds PASSED
test_netmiko_show.py::test_clear_buffer PASSED
test_netmiko_show.py::test_enable_mode PASSED
test_netmiko_show.py::test_disconnect PASSED
test_netmiko_show.py::test_disconnect_no_enable PASSED

=========================== short test summary info ============================
SKIPPED [1] /home/kbyers/netmiko/tests/test_netmiko_show.py:68: <Skipped instance>
SKIPPED [1] /home/kbyers/netmiko/tests/test_netmiko_show.py:86: <Skipped instance>
SKIPPED [1] /home/kbyers/netmiko/tests/test_netmiko_show.py:108: <Skipped instance>
======================== 17 passed, 3 skipped in 31.71s ========================
============================= test session starts ==============================
platform linux -- Python 3.8.5, pytest-5.1.2, py-1.10.0, pluggy-0.13.1 -- /home/kbyers/netmiko/.venv/bin/python3.8
cachedir: .pytest_cache
rootdir: /home/kbyers/netmiko, inifile: setup.cfg
plugins: pylama-7.7.1
collecting ... collected 2 items

test_netmiko_save.py::test_save_base PASSED
test_netmiko_save.py::test_disconnect PASSED

============================== 2 passed in 5.75s ===============================
============================= test session starts ==============================
platform linux -- Python 3.8.5, pytest-5.1.2, py-1.10.0, pluggy-0.13.1 -- /home/kbyers/netmiko/.venv/bin/python3.8
cachedir: .pytest_cache
rootdir: /home/kbyers/netmiko, inifile: setup.cfg
plugins: pylama-7.7.1
collecting ... collected 2 items

test_netmiko_cmd_verify.py::test_cmd_verify_decorator PASSED
test_netmiko_cmd_verify.py::test_send_command_global_cmd_verify PASSED

============================== 2 passed in 2.77s ===============================
============================= test session starts ==============================
platform linux -- Python 3.8.5, pytest-5.1.2, py-1.10.0, pluggy-0.13.1 -- /home/kbyers/netmiko/.venv/bin/python3.8
cachedir: .pytest_cache
rootdir: /home/kbyers/netmiko, inifile: setup.cfg
plugins: pylama-7.7.1
collecting ... collected 10 items

test_netmiko_config.py::test_ssh_connect PASSED
test_netmiko_config.py::test_enable_mode PASSED
test_netmiko_config.py::test_config_mode PASSED
test_netmiko_config.py::test_exit_config_mode PASSED
test_netmiko_config.py::test_config_set PASSED
test_netmiko_config.py::test_config_set_longcommand PASSED
test_netmiko_config.py::test_config_hostname PASSED
test_netmiko_config.py::test_config_from_file PASSED
test_netmiko_config.py::test_config_error_pattern PASSED
test_netmiko_config.py::test_disconnect PASSED

============================== 10 passed in 7.06s ==============================
============================= test session starts ==============================
platform linux -- Python 3.8.5, pytest-5.1.2, py-1.10.0, pluggy-0.13.1 -- /home/kbyers/netmiko/.venv/bin/python3.8
cachedir: .pytest_cache
rootdir: /home/kbyers/netmiko, inifile: setup.cfg
plugins: pylama-7.7.1
collecting ... collected 1 item

test_netmiko_config_acl.py::test_large_acl PASSED

============================== 1 passed in 7.01s ===============================
============================= test session starts ==============================
platform linux -- Python 3.8.5, pytest-5.1.2, py-1.10.0, pluggy-0.13.1 -- /home/kbyers/netmiko/.venv/bin/python3.8
cachedir: .pytest_cache
rootdir: /home/kbyers/netmiko, inifile: setup.cfg
plugins: pylama-7.7.1
collecting ... collected 6 items

test_netmiko_tcl.py::test_tcl_put PASSED
test_netmiko_tcl.py::test_remote_space_available PASSED
test_netmiko_tcl.py::test_verify_space_available_put PASSED
test_netmiko_tcl.py::test_remote_file_size PASSED
test_netmiko_tcl.py::test_md5_methods PASSED
test_netmiko_tcl.py::test_disconnect PASSED

============================== 6 passed in 12.07s ==============================
============================= test session starts ==============================
platform linux -- Python 3.8.5, pytest-5.1.2, py-1.10.0, pluggy-0.13.1 -- /home/kbyers/netmiko/.venv/bin/python3.8
cachedir: .pytest_cache
rootdir: /home/kbyers/netmiko, inifile: setup.cfg
plugins: pylama-7.7.1
collecting ... collected 12 items

test_netmiko_scp.py::test_scp_put PASSED
test_netmiko_scp.py::test_remote_space_available PASSED
test_netmiko_scp.py::test_local_space_available PASSED
test_netmiko_scp.py::test_verify_space_available_put PASSED
test_netmiko_scp.py::test_remote_file_size PASSED
test_netmiko_scp.py::test_md5_methods PASSED
test_netmiko_scp.py::test_disconnect PASSED
test_netmiko_scp.py::test_verify_space_available_get PASSED
test_netmiko_scp.py::test_scp_get PASSED
test_netmiko_scp.py::test_md5_methods_get PASSED
test_netmiko_scp.py::test_disconnect_get PASSED
test_netmiko_scp.py::test_file_transfer PASSED

============================= 12 passed in 22.16s ==============================
============================= test session starts ==============================
platform linux -- Python 3.8.5, pytest-5.1.2, py-1.10.0, pluggy-0.13.1 -- /home/kbyers/netmiko/.venv/bin/python3.8
cachedir: .pytest_cache
rootdir: /home/kbyers/netmiko, inifile: setup.cfg
plugins: pylama-7.7.1
collecting ... collected 1 item

test_netmiko_autodetect.py::test_ssh_connect cisco_ios
PASSED

============================== 1 passed in 7.14s ===============================
